spec/system/admin/reports/orders_and_fulfillment_spec.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+require "system_helper"
+
+describe "Orders And Fulfillment" do
+  include AuthenticationHelper
+  include WebHelper
+
+  describe "reports" do
+    before do
+      login_as_admin
+      visit admin_reports_path
+    end
+
+    let(:bill_address1) { create(:address, lastname: "ABRA") }
+    let(:bill_address2) { create(:address, lastname: "KADABRA") }
+    let(:distributor_address) {
+      create(:address, address1: "distributor address", city: 'The Shire', zipcode: "1234")
+    }
+    let(:distributor) { create(:distributor_enterprise, address: distributor_address) }
+    let(:order1) {
+      create(:completed_order_with_totals, line_items_count: 0, distributor: distributor,
+                                           bill_address: bill_address1)
+    }
+    let(:order2) {
+      create(:completed_order_with_totals, line_items_count: 0, distributor: distributor,
+                                           bill_address: bill_address1)
+    }
+    let(:supplier) { create(:supplier_enterprise, name: "Supplier") }
+    let(:product) { create(:simple_product, name: "Product", supplier: supplier ) }
+    let(:variant) { create(:variant, product: product, unit_description: "Big") }
+
+    before do
+      Timecop.travel(Time.zone.local(2022, 4, 25, 14, 0, 0)) { order1.finalize! }
+      Timecop.travel(Time.zone.local(2022, 4, 25, 15, 0, 0)) { order2.finalize! }
+
+      create(:line_item_with_shipment, variant: variant, quantity: 1, order: order1)
+      create(:line_item_with_shipment, variant: variant, quantity: 2, order: order2)
+    end
+
+    describe "Order Cycle Customer Totals" do
+      before do
+        click_link "Order Cycle Customer Totals"
+      end
+
+      it "displays the report" do
+        find('#q_completed_at_gt').click
+        select_date_from_datepicker Time.zone.at(order1.completed_at - 1.day)
+
+        find('#q_completed_at_lt').click
+        select_date_from_datepicker Time.zone.at(order2.completed_at + 1.day)
+
+        click_button 'Go'
+
+        rows = find("table.report__table").all("thead tr")
+        table = rows.map { |r| r.all("th").map { |c| c.text.strip } }
+        expect(table).to eq([
+                              [I18n.t("report_header_hub"),
+                               I18n.t("report_header_customer"),
+                               I18n.t("report_header_email"),
+                               I18n.t("report_header_phone"),
+                               I18n.t("report_header_producer"),
+                               I18n.t("report_header_product"),
+                               I18n.t("report_header_variant"),
+                               I18n.t("report_header_quantity"),
+                               I18n.t("report_header_item_price", currency: currency_symbol),
```

Comment:
Oh, and by testing against the rendered text, we also make sure that variables like the currency have been passed on correctly. It's easy to copy and paste incorrect code.